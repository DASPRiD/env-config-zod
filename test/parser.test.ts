import { assert, expect, it } from 'vitest';
import { z } from 'zod';
import { EnvConfigError, parseEnvConfig } from '../src/index.js';

it("should parse simple config", () => {
    const envConfig = parseEnvConfig(
        {
            foo: z.string(),
        },
        "",
        {
            FOO: "bar",
        },
    );

    expect(envConfig).toEqual({
        foo: "bar",
    });
});

it("should parse nested configs", () => {
    const envConfig = parseEnvConfig(
        {
            foo: {
                bar: z.string(),
            },
        },
        "",
        {
            FOO_BAR: "baz",
        },
    );

    expect(envConfig).toEqual({
        foo: {
            bar: "baz",
        },
    });
});

it("should convert camel case to constant case", () => {
    const envConfig = parseEnvConfig(
        {
            fooBar: z.string(),
        },
        "",
        {
            FOO_BAR: "baz",
        },
    );

    expect(envConfig).toEqual({
        fooBar: "baz",
    });
});

it("should emit combined issues", () => {
    try {
        parseEnvConfig(
            {
                foo: z.string(),
                bar: z.string(),
            },
            "",
            {},
        );

        assert.fail("Expected error was not thrown");
    } catch (error) {
        expect(error).toBeInstanceOf(EnvConfigError);
        expect((error as EnvConfigError).issues).toMatchSnapshot();
    }
});

it("should collect nested issues", () => {
    try {
        parseEnvConfig(
            {
                foo: z.string(),
                bar: {
                    bat: z.string(),
                },
                baz: {
                    bat: z.string(),
                },
            },
            "",
            {},
        );

        assert.fail("Expected error was not thrown");
    } catch (error) {
        expect(error).toBeInstanceOf(EnvConfigError);
        expect((error as EnvConfigError).issues).toMatchSnapshot();
    }
});
