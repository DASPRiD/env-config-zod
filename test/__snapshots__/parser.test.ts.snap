// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should collect nested issues 1`] = `
[
  {
    "issues": [
      {
        "code": "invalid_type",
        "expected": "string",
        "message": "Required",
        "path": [],
        "received": "undefined",
      },
    ],
    "name": "FOO",
  },
  {
    "issues": [
      {
        "code": "invalid_type",
        "expected": "string",
        "message": "Required",
        "path": [],
        "received": "undefined",
      },
    ],
    "name": "BAR_BAT",
  },
  {
    "issues": [
      {
        "code": "invalid_type",
        "expected": "string",
        "message": "Required",
        "path": [],
        "received": "undefined",
      },
    ],
    "name": "BAZ_BAT",
  },
]
`;

exports[`should emit combined issues 1`] = `
[
  {
    "issues": [
      {
        "code": "invalid_type",
        "expected": "string",
        "message": "Required",
        "path": [],
        "received": "undefined",
      },
    ],
    "name": "FOO",
  },
  {
    "issues": [
      {
        "code": "invalid_type",
        "expected": "string",
        "message": "Required",
        "path": [],
        "received": "undefined",
      },
    ],
    "name": "BAR",
  },
]
`;
